### PR TITLE
Add ability to set default checked level

### PR DESCRIPTION
**Pull request by jez**
### Motivation

From @ptarjan:

> The two most asked for features are "how do I turn the whole thing off" and
> "how do I turn the whole thing off in production".

By giving people the ability to set a default checked level, people can opt out
of the runtime overhead. Note that in the presence of performance constraints,
just setting a T::Configuration call_validation handler is not enough. The
method will still have the runtime overhead of type checking. The only current
option is to `.checked(:never)` or `.checked(:tests)` the entire method.

I think we should also let `.soft` and `.checked` be chained, because their
behavior is non-orthogonal. `.soft` configures what options are passed to the
T::Configuration handler when it **does** fail, and whereas `.checked` controls
whether it has the ability to fail at all.

Before we add this into the documentation, I think we should be absolutely
clear about the tradeoffs here, namely that runtime checking is **incredibly**
important to dealing with untyped code.

### Test plan

See included automated tests.

### CODE REVIEW

File: gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb [hunk 1]
```diff
@@ -249,7 +249,7 @@ def finalize!
         decl.bind = nil
       end
       if decl.checked.equal?(ARG_NOT_PROVIDED)
-        decl.checked = :always
+        decl.checked = T::Private::RuntimeLevels.default_checked_level
```
Comment (jez) [inline gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb]: As currently implemented,

```ruby
sig {returns(...).soft(...).checked(:never)}
```

is an error, but

```ruby
T::Configuration.default_checked_level = :never
sig {returns(...).soft}
```

is not an error, but the `.soft` will be ignored. This is probably a bug, and I think i'd like to fix it.

**Comment by jez** [comment]
We might also want to give this a scarier name. Some of them that were suggested before:

- `disable_runtime_safety_checks`
- `global_disable_runtime_errors`
- `global_disable_runtime_checks`

**Comment by jez** [comment]
- [ ] Note to self—I forgot to add a test that `.checked(:always)` raises type errors when the default is `:never`. this test will have to be skipped until we remove the raise from `def checked`.

**Comment by XrXr** [comment]
Here's another naming suggestion `default_runtime_safety_level`. Mentions safety and default.

File: gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb (hunk 1 above)
Comment (jez) [reply gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb]: This invariant is easier to enforce on my other branch that re-enables .checked. I'll add it there.

**Pull request merged**